js_modules/dagster-ui/packages/ui-core/src/sensors/SensorRoot.tsx

```diff
@@ -77,61 +86,87 @@ export const SensorRoot = ({repoAddress}: {repoAddress: RepoAddress}) => {
       }}
     />
   );
+
+  if (!data && loading) {
+    return (
+      <Box margin={{top: 32}} flex={{direction: 'row', alignItems: 'center', gap: 16}}>
+        <Spinner purpose="body-text" />
+        <div style={{color: colorTextLight()}}>Loading sensor…</div>
+      </Box>
+    );
+  }
+
+  if (!data || data.sensorOrError.__typename === 'SensorNotFoundError') {
+    return (
+      <Box padding={{vertical: 32}}>
+        <NonIdealState
+          icon="error"
+          title={`Could not find sensor \`${sensorName}\` in definitions for \`${repoAddress.name}\``}
+        />
+      </Box>
+    );
+  }
+
+  const {sensorOrError} = data;
+  if (sensorOrError.__typename === 'PythonError') {
+    return <PythonErrorInfo error={sensorOrError} />;
+  }
+
+  if (sensorOrError.__typename === 'UnauthorizedError') {
+    return <Redirect to="/overview/sensors" />;
+  }
+
+  const {instance} = data;
+
+  if (sensorOrError.sensorType === SensorType.AUTOMATION_POLICY) {
+    const assetDaemonStatus = instance.daemonHealth.ampDaemonStatus;
+    return (
+      <Page>
+        <SensorDetails
+          repoAddress={repoAddress}
+          sensor={sensorOrError}
+          daemonHealth={assetDaemonStatus.healthy}
+          refreshState={refreshState}
+        />
+        <SensorPageAutomaterialize
+          repoAddress={repoAddress}
+          sensor={sensorOrError}
+          daemonStatus={assetDaemonStatus}
+          loading={loading}
+        />
+      </Page>
+    );
+  }
+
+  const sensorDaemonStatus = instance.daemonHealth.sensorDaemonStatus;
+
   return (
-    <Loading queryResult={queryResult} allowStaleData={true}>
```

Comment:
The `Loading` componnt doesn't do much for us in general, and it often just gets in the way. I'm unwrapping it here.